Changelog 24.11.0 (#5540)
conda/conda-build: CHANGELOG.md

[//]: # (current developments)

## 24.11.0 (2024-11-15)

### Enhancements

> * Introduce `--package-format` as command line argument. (#4890 via #5209)

Author: ```suggestion
* Introduce `--package-format` as a command line argument. (#4890 via #5209)
```

> * `--package-format` introduced as command line argument. (#4890 via #5209)

Author: ```suggestion
* Introduce `--package-format` as command line argument. (#4890 via #5209)
```

  * This takes precedence over default value and `condarc`.
  * Normalization occurs so `1`, `"1"`, `tar.bz2`,`.tar.bz2`, `2`, `"2"`, `conda`, `.conda` are all recognized and mapped appropriately.
  * Other options are rejected.

> * Add  support for CEP-17 that allows specifying the location of the site-packages directory with the `python_site_packages_path` build option for any packages named `python`. (#5502)

Author: ```suggestion
* Add  support for [CEP-17](https://github.com/conda/ceps/blob/main/cep-0017.md) that allows specifying the location of the site-packages directory with the `python_site_packages_path` build option for any packages named `python`. (#5502)
```

### Bug fixes

* Fix regex for Jinja2 `set` / `for` statements to be more specific. (#5514)
* Fix `ruamel.yaml` usage to use supported APIs. (#5517)

> * Fix bug in docs build by pinning `conda-sphinx-theme` version to 0.2.2. (#5518)

Reviewer: Does this belong to `# Docs` better?

Reviewer 2: There is a bug in rever that miscounts things, was fixed in https://github.com/regro/rever/pull/266 but that was never released, so you likely used a version that miscounts contributions basically

Reviewer 2: You can redo the run if you want with that new version

> * Fixed regex for Jinja2 `set` / `for` statements to be more specific. (#5514)
> * Fixed ruamel.yaml usage to use supported APIs. (#5517)
> * Fixed bug in docs build by pinning conda-sphinx-theme to 0.2.2. (#5518)
> * Fixed bug variant variables were not defined for the first parsing pass of a recipe. (#5528)
> * Fixed a bug where variants were incorrectly found as being used when they matched a leading substring of
>   another variant. (#5535)
> * Fixed a bug where variants were not found when variables were used in ``pin_*`` statements. (#5535)

Author: ```suggestion
* Fix regex for Jinja2 `set` / `for` statements to be more specific. (#5514)
* Fix `ruamel.yaml` usage to use supported APIs. (#5517)
* Fix bug in docs build by pinning `conda-sphinx-theme` version to 0.2.2. (#5518)
* Fix bug variant variables were not defined for the first parsing pass of a recipe. (#5528)
* Fix a bug where variants were incorrectly found as being used when they matched a leading substring of
  another variant. (#5535)
* Fix a bug where variants were not found when variables were used in `pin_*` statements. (#5535)
```
